modules/core/test/test_rotatedrect.cpp

```diff
@@ -105,4 +105,82 @@ int Core_RotatedRectConstructorTest::validate_test_results( int )
 
 TEST(Core_RotatedRect, three_point_constructor) { Core_RotatedRectConstructorTest test; test.safe_run(); }
 
+TEST(Core_RotatedRect, intersection)
+{
+    float scaleFactor = 1000;//to challenge the numericalScalingFactor in the algorithm
+    cv::RectanglesIntersectTypes intersectionResult = cv::RectanglesIntersectTypes::INTERSECT_NONE;
+    std::vector<cv::Point2f> intersection;
+    double intersectionArea = 0;
+    cv::RotatedRect r1 = cv::RotatedRect(cv::Point2f(.5f, .5f)*scaleFactor, cv::Size2f(1.f, 1.f)*scaleFactor, 0);
+    cv::RotatedRect r2;
+
+    r2 = cv::RotatedRect(cv::Point2f(-2.f, -2.f)*scaleFactor, cv::Size2f(1.f, 1.f)*scaleFactor, 0);
+    intersectionResult = (cv::RectanglesIntersectTypes) cv::rotatedRectangleIntersection(r1, r2, intersection);
+    intersectionArea = (intersection.size() <= 2) ? 0. : cv::contourArea(intersection);
+    ASSERT_EQ(intersectionResult, cv::RectanglesIntersectTypes::INTERSECT_NONE);
```

Comment:
> _EQ

Signature for `_EQ` is `ASSERT_EQ(expected_reference, actual_result);`.
Correct order of parameters are required to emit valid error messages.

Reference: https://github.com/opencv/opencv/blob/4.0.0/modules/ts/include/opencv2/ts/ts_gtest.h#L8196-L8200

```
GTEST_API_ AssertionResult EqFailure(const char* expected_expression,
                                     const char* actual_expression,
                                     const std::string& expected_value,
                                     const std::string& actual_value,
                                     bool ignoring_case);
```

Reply:
:+1: